lib/src/kbasesearchengine/system/ObjectTypeParsingRulesUtils.java

```diff
@@ -0,0 +1,150 @@
+package kbasesearchengine.system;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.Map;
+
+import kbasesearchengine.common.ObjectJsonPath;
+import kbasesearchengine.parse.ObjectParseException;
+import kbasesearchengine.system.ObjectTypeParsingRules.Builder;
+import kbasesearchengine.tools.Utils;
+import us.kbase.common.service.UObject;
+
+/** Utilities for creating {@link ObjectTypeParsingRules} from various data sources.
+ * @author gaprice@lbl.gov
+ *
+ */
+public class ObjectTypeParsingRulesUtils {
+
+    //TODO CODE not sure if throwing ObjectParseException makes sense here
+    //TODO TEST
+
+    /** Create an ObjectTypeParsingRules instance from a file.
+     * 
+     * TODO document the file structure.
+     * @param file the file containing the parsing rules.
+     * @return a new set of parsing rules.
+     * @throws ObjectParseException if the file contains erroneous parsing rules.
+     * @throws IOException if an IO error occurs reading the file.
+     */
+    public static ObjectTypeParsingRules fromFile(final File file) 
+            throws ObjectParseException, IOException {
+        try (final InputStream is = new FileInputStream(file)) {
+            return fromStream(is, file.toString());
+        }
+    }
+
+    private static ObjectTypeParsingRules fromStream(InputStream is, String sourceInfo) 
+            throws IOException, ObjectParseException {
+        @SuppressWarnings("unchecked")
+        Map<String, Object> obj = UObject.getMapper().readValue(is, Map.class);
+        return fromObject(obj, sourceInfo);
+    }
+
+    private static ObjectTypeParsingRules fromObject(
```

Comment:
The tests for this class have failure cases implemented but the exception lines in this method appear not to be covered in the jacoco output (KBaseSearchEngine/test_local/workdir/test-reports/html/kbasesearchengine.system/ObjectTypeParsingRulesUtils.java.html#L52)

Reply:
I intentionally didn't add any unit tests for this class because the pr was concentrating on the refactor of OTPR, not the utility method to load one from a file. It's no worse than before though. We'll add unit tests later.